compiler/rustc_macros/src/diagnostics/utils.rs

```diff
@@ -601,6 +612,9 @@ pub(super) struct SubdiagnosticVariant {
     pub(super) kind: SubdiagnosticKind,
     pub(super) slug: Option<Path>,
     pub(super) no_span: bool,
+    /// A subdiagnostic can have a raw_label field, e.g. `#[help("some text")]`.
+    /// if `slug` is None, this field need to be set.
+    pub(super) raw_label: Option<LitStr>,
```

Comment:
Can we use an enum for `slug` that has a `message` or a `slug`, to capture the mutual exclusive-ness of these?